packages/swapper/src/swappers/ButterSwap/xhr.ts

```diff
@@ -0,0 +1,179 @@
+import { fromAssetId, solanaChainId } from '@shapeshiftoss/caip'
+import type { Asset } from '@shapeshiftoss/types'
+import { chainIdToFeeAssetId } from '@shapeshiftoss/utils'
+import type { Result } from '@sniptt/monads'
+import { Err, Ok } from '@sniptt/monads'
+import { zeroAddress } from 'viem'
+
+import type { SwapErrorRight } from '../../types'
+import { TradeQuoteError } from '../../types'
+import { makeSwapErrorRight } from '../../utils'
+import type { BridgeInfo, BridgeInfoApiResponse, BuildTxResponse, RouteResponse } from './types'
+import { butterHistoryService } from './utils/butterSwapHistoryService'
+import { butterService } from './utils/butterSwapService'
+import { chainIdToButterSwapChainId } from './utils/helpers'
+
+export enum ButterSwapErrorCode {
+  ParameterError = 2000,
+  ChainNotSupported = 2001,
+  TokenNotSupported = 2002,
+  NoRouteFound = 2003,
+  InsufficientLiquidity = 2004,
+  SlippageOutOfRange = 2005,
+  InsufficientAmount = 2006,
+  InvalidAddress = 2007,
+}
+
+type ButterSwapPromise<T> = Promise<Result<T, SwapErrorRight>>
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-route
+ */
+export type GetButterRouteArgs = {
+  sellAsset: Asset
+  buyAsset: Asset
+  sellAmountCryptoBaseUnit: string
+  slippage: string
+  affiliate?: string
+}
+
+const SOLANA_NATIVE_ADDRESS = 'So11111111111111111111111111111111111111112'
+
+export const getButterRoute = async ({
+  sellAsset,
+  buyAsset,
+  sellAmountCryptoBaseUnit,
+  slippage,
+  affiliate,
+}: GetButterRouteArgs): ButterSwapPromise<RouteResponse> => {
+  const butterFromChainId = chainIdToButterSwapChainId(sellAsset.chainId)
+  const butterToChainId = chainIdToButterSwapChainId(buyAsset.chainId)
+  if (!butterFromChainId || !butterToChainId) {
+    return Err(
+      makeSwapErrorRight({
+        message: '[getButterRoute] Unsupported chainId',
+        code: TradeQuoteError.UnsupportedChain,
+      }),
+    )
```

Comment:
Ok, that's where the chain support checks comes from, so it *is* implemented
We shouldn't handle this here though, as we shouldn't ever hit this endpoint for unsupported chains. 
Checks should be done at the beginning of `getTradeQuote` / `getTradeRate` with checks against supported ChainIds, before any other logic (fine to also keep here for the sake of narrowing, even though this won't be hit)